core/services/registrysyncer/syncer.go

```diff
@@ -276,6 +276,11 @@ func (s *registrySyncer) Sync(ctx context.Context, isInitialSync bool) error {
 			return err
 		}
 
+		err = reader.Start(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
Shouldn't it be started in `func (s *registrySyncer) Start(ctx context.Context) error {`

Reply:
`func (s *registrySyncer) Start(ctx context.Context) error` calls `func (s *registrySyncer) syncLoop()` that calls `func (s *registrySyncer) Sync(ctx context.Context, isInitialSync bool) error` and ends up executing the initialization of the CR. 

I added the start of the CR service right after this initialization

Reply:
What do you mean by `this problem`? 

There's a note in the `newReader` func that says why this is achieved this way: 
> // NOTE: this can't be called while initializing the syncer and needs to be called in the sync loop.
// This is because Bind() makes an onchain call to verify that the contract address exists, and if
// called during initialization, this results in a "no live nodes" error.

Reply:
I added these lines as we are now enforcing CR service to be in `Started` state before calling the methods. This `syncer` was not starting the service previously and that's why we got some tests failing but now they are passing with this change

Reply:
> What do you mean by this problem?

If all dependencies are `Started` before anything that uses them, then there is no need to be defensive about checking internally if `Start` has been called. But I see that there is a good reason to defer in this case.